unittests/misc_tests.cpp

```diff
@@ -832,47 +832,37 @@ BOOST_AUTO_TEST_CASE(transaction_metadata_test) { try {
       packed_transaction pkt(trx, packed_transaction::compression_type::none);
       packed_transaction pkt2(trx, packed_transaction::compression_type::zlib);
 
-      transaction_metadata_ptr mtrx = std::make_shared<transaction_metadata>( std::make_shared<packed_transaction>( trx, packed_transaction::compression_type::none) );
-      transaction_metadata_ptr mtrx2 = std::make_shared<transaction_metadata>( std::make_shared<packed_transaction>( trx, packed_transaction::compression_type::zlib) );
+      packed_transaction_ptr ptrx = std::make_shared<packed_transaction>( trx, packed_transaction::compression_type::none);
+      packed_transaction_ptr ptrx2 = std::make_shared<packed_transaction>( trx, packed_transaction::compression_type::zlib);
 
       BOOST_CHECK_EQUAL(trx.id(), pkt.id());
       BOOST_CHECK_EQUAL(trx.id(), pkt2.id());
-      BOOST_CHECK_EQUAL(trx.id(), mtrx->id());
-      BOOST_CHECK_EQUAL(trx.id(), mtrx2->id());
+      BOOST_CHECK_EQUAL(trx.id(), ptrx->id());
+      BOOST_CHECK_EQUAL(trx.id(), ptrx2->id());
 
       named_thread_pool thread_pool( "misc", 5 );
 
-      transaction_metadata::start_recover_keys( mtrx, thread_pool.get_executor(), test.control->get_chain_id(), fc::microseconds::maximum() );
-      transaction_metadata::start_recover_keys( mtrx2, thread_pool.get_executor(), test.control->get_chain_id(), fc::microseconds::maximum() );
+      auto fut = transaction_metadata::start_recover_keys( ptrx, thread_pool.get_executor(), test.control->get_chain_id(), fc::microseconds::maximum() );
+      auto fut2 = transaction_metadata::start_recover_keys( ptrx2, thread_pool.get_executor(), test.control->get_chain_id(), fc::microseconds::maximum() );
 
       // no-op
```

Comment:
These wouldn't be no-ops anymore right? It would actually be starting another key recovery process now.

Reply:
Correct, so no-op in terms of nothing bad happens. I'll change it to:
`// start another key recovery on same transaction_metadata`